azext_iot/operations/events3/_events.py

```diff
@@ -142,7 +150,18 @@ def _output_msg_kpi(msg):
             if app_prop:
                 event_source['event']['properties']['application'] = unicode_binary_map(app_prop)
 
-        six.print_(yaml.dump(event_source, default_flow_style=False), flush=True)
+        if output and output == 'json':
+            try:
+                if event_source['event']['payload']:
+                    event_source['event']['payload'] = json.loads(event_source['event']['payload'].replace('\\n', '').replace('\\r', ''))
+            except Exception:  # pylint: disable=broad-except
+                # We don't want to crash the monitor if JSON parsing fails.
+                six.print_('Message payload is not valid JSON.')
```

Comment:
I'm catching all JSON errors so we don't crash the terminal.

Reply:
The try/except in Python is extremely efficient but not if the expectation is wrong most of the time (exception gets triggered more often than not). In the current state, if the output setting is json and you are receiving telemetry that isn't compatible with that, you are taking a notable performance hit since every message triggers the exception.

This six.print_ msg should indicate to switch to yaml output for raw data and it should be clearly noted in params and help text to not use output JSON if you don't expect JSON.

Ideally, maybe in a future state the incoming messages content-type system property should be set with say application/json, then on the receiving end the message payload goes through the respective conversion process and if that fails, fallback to default (raw text) and output the invalid json notice (if content-type is application/json the expectation is for valid json).

Reply:
The default output is json. So as of now this code will try to output as json. 

Maybe we need a content-type property or a flag that indicates the input type and only parse the payload as json if that is set to json. 

The current code assumes that if you want json output that means you have json input which isn’t always the case. 

Let me revise this and see if we can inspect and determine payload type without having to loads.

Reply:
Yes this what I'm trying to describe in the last paragraph. There is a system property called content-type. You can test setting it via send-d2c-message by including the property like --props '$.ct=application/json'

Reply:
This on send-d2c-message:

`--props $.ct=application/json`

Is automatically translated to this:
```
"properties": {
            "system": {
                "content_type": "application/json"
            }
        }
```